### PR TITLE
cayley integration, update docs

### PR DESCRIPTION
**Pull request by dwhitena**
- integrate cayley functionality in DB.
- small change in the mongo DB functionality to handle the changes.
- run update docs script.

### CODE REVIEW

File: db/cayley.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	db.graphHandle = store
+	return nil
+}
+
+// GraphHandle returns the Cayley graph handle for graph interactions.
+func (db *DB) GraphHandle(context interface{}) (*cayley.Handle, error) {
+	if db.graphHandle != nil {
+		return db.graphHandle, nil
+	}
+	return nil, errors.New("Graph handle not initialized.")
```
Comment (wyattjoh) [inline db/cayley.go]: Extract error to global.

File: db/cayley.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+// CloseCayley closes a graph handle value.
+func (db *DB) CloseCayley(context interface{}) {
+	db.graphHandle.Close()
```
Comment (wyattjoh) [inline db/cayley.go]: Does this return an error? If so it should return.

File: db/mongo.go [hunk 3]
```diff
@@ -72,7 +72,7 @@ func NewMGO(context interface{}, name string) (*DB, error) {
 	ses := db.ses.Copy()
 	mdb := ses.DB(db.dbName)
 
-	return &DB{mdb, ses}, nil
+	return &DB{database: mdb, session: ses}, nil
```
Comment (wyattjoh) [inline db/mongo.go]: Should break this into another line instead of a single line.

Comment (dwhitena) [inline db/mongo.go]: Fixed here https://github.com/ardanlabs/kit/pull/35/commits/a821f60defc8de69862ddd5f317a8fa43843523d

**Pull request merged**